### PR TITLE
scheduler: use move-hot-write-leader operator

### PR DESCRIPTION
**Pull request by lhy1024**
### What problem does this PR solve?

Issue Number: Close https://github.com/tikv/pd/issues/7848
### What is changed and how does it work?

```commit-message
```

### Check List

Tests

- Unit test
- [x] manual test with 6 tikv
    - tpcc 
image
    - oltp_read_write 
image
    - oltp_write_only
image
    - oltp_update_index
image
    - oltp_update_non_index
image
    - oltp_insert
image

### Release note

```release-note
None.
```

### CODE REVIEW

File: pkg/schedule/schedulers/hot_region_test.go [hunk 1]
```diff
@@ -533,11 +535,14 @@ func checkHotWriteRegionScheduleByteRateOnly(re *require.Assertions, enablePlace
...
 		clearPendingInfluence(hb.(*hotScheduler))
-		re.Equal(4, op.Len())
 		if op.RegionID() == 2 {
+			re.Equal("move-hot-write-peer", op.Desc())
+			re.Equal(4, op.Len())
 			// peer in store 1 of the region 2 can transfer to store 5 or store 6 because of the label
 			operatorutil.CheckTransferPeerWithLeaderTransferFrom(re, op, operator.OpHotRegion, 1)
 		} else {
+			re.Equal("move-hot-write-leader", op.Desc())
+			re.Equal(5, op.Len())
```
Comment (rleungx) [inline pkg/schedule/schedulers/hot_region_test.go:545]: why is the length different?

Comment (lhy1024) [reply pkg/schedule/schedulers/hot_region_test.go:545]: https://github.com/tikv/pd/issues/7872 Move leader without joint will appear as an extra transfer leader, I'll try to adjust the test firstly to make it look more reasonable

File: pkg/schedule/schedulers/hot_region_test.go [hunk 2]
```diff
@@ -535,14 +532,11 @@ func checkHotWriteRegionScheduleByteRateOnly(re *require.Assertions, enablePlace
...
 		clearPendingInfluence(hb.(*hotScheduler))
+		re.Equal(5, op.Len())
 		if op.RegionID() == 2 {
-			re.Equal("move-hot-write-peer", op.Desc())
-			re.Equal(4, op.Len())
 			// peer in store 1 of the region 2 can transfer to store 5 or store 6 because of the label
-			operatorutil.CheckTransferPeerWithLeaderTransferFrom(re, op, operator.OpHotRegion, 1)
+			operatorutil.CheckTransferPeerWithLeaderTransfer(re, op, operator.OpHotRegion, 1, 0)
 		} else {
-			re.Equal("move-hot-write-leader", op.Desc())
```
Comment (rleungx) [inline pkg/schedule/schedulers/hot_region_test.go]: no need it?

**Review by rleungx — APPROVED** [review]

File: pkg/utils/operatorutil/operator_check.go [hunk 3]
```diff
@@ -88,32 +104,36 @@ func CheckTransferLearner(re *require.Assertions, op *operator.Operator, kind op
 // CheckTransferPeerWithLeaderTransfer checks if the operator is to transfer
 // peer between the specified source and target stores and it meanwhile
 // transfers the leader out of source store.
+// If targetID is 0, it means the operator is to transfer peer to any store.
 func CheckTransferPeerWithLeaderTransfer(re *require.Assertions, op *operator.Operator, kind operator.OpKind, sourceID, targetID uint64) {
```
[the remark this answers is not in the document]
Comment (lhy1024) [reply pkg/utils/operatorutil/operator_check.go:108]: Maybe I will refactor them in another pr.

File: metrics/grafana/pd.json [hunk 4]
```diff
@@ -5881,6 +5881,21 @@
               "intervalFactor": 1,
               "legendFormat": "store-{{store}}-in",
               "refId": "B"
+            },
+            {
+              "expr": "- sum(delta(pd_scheduler_hot_region_direction{k8s_cluster=\"$k8s_cluster\", tidb_cluster=\"$tidb_cluster\", store=~\"$store\",type=\"move-leader\",direction=\"out\",rw=\"write\"}[1m]))by (store)",
```
[the remark this answers is not in the document]
Comment (lhy1024) [reply metrics/grafana/pd.json:5886]: It should be negative, because its direction is out

[the remark this answers is not in the document]
Comment (HuSharp) [reply metrics/grafana/pd.json:5886]: got it.

**Review by HuSharp — APPROVED** [review]

**Pull request merged**